azurerm/resource_arm_application_insights_webtests.go

```diff
@@ -0,0 +1,324 @@
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmApplicationInsightsWebTestsRead(d, meta)
+}
+
+func resourceArmApplicationInsightsWebTestsRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).appInsightsWebTestsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] Reading AzureRM Application Insights WebTests '%s'", id)
+
+	resGroup := id.ResourceGroup
+	name := id.Path["webtests"]
+
+	resp, err := client.Get(ctx, resGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on AzureRM Application Insights WebTests '%s': %+v", name, err)
+	}
+
+	log.Printf("[DEBUG] AzureRM Application Insights WebTests name '%s'", name)
+
+	appInsightsId := ""
+	tags := resp.Tags
+	for i := range tags {
+		if strings.HasPrefix(i, "hidden-link") {
+			appInsightsId = strings.Split(i, ":")[1]
+		}
+	}
+	d.Set("application_insights_id", appInsightsId)
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resGroup)
+
+	if location := resp.Location; location != nil {
+		d.Set("location", azureRMNormalizeLocation(*location))
+	}
+
+	if kind := resp.Kind; kind != "" {
+		d.Set("kind", resp.Kind)
+	}
+
+	if props := resp.WebTestProperties; props != nil {
+		d.Set("synthetic_monitor_id", props.SyntheticMonitorID)
+		d.Set("description", props.Description)
+		d.Set("enabled", props.Enabled)
+		d.Set("frequency", props.Frequency)
+		d.Set("timeout", props.Timeout)
+		d.Set("retry_enabled", props.RetryEnabled)
+		d.Set("configuration", props.Configuration)
+		d.Set("provisioning_state", props.ProvisioningState)
+		d.Set("test_configuration", props.Configuration.WebTest)
+
+		if err := d.Set("geo_locations", flattenGeoLocations(props.Locations)); err != nil {
+			return fmt.Errorf("Error setting `geo_locations`: %+v", err)
+		}
+	}
+
+	log.Printf("[DEBUG] AzureRM Application Insights WebTests synetheticmonitorid '%s'", d.Get("synthetic_monitor_id"))
+
+	flattenAndSetTags(d, resp.Tags)
+
+	return nil
+}
+
+func resourceArmApplicationInsightsWebTestsDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).appInsightsWebTestsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["webtests"]
+
+	log.Printf("[DEBUG] Deleting AzureRM Application Insights WebTest '%s' (resource group '%s')", name, resGroup)
+
+	resp, err := client.Delete(ctx, resGroup, name)
+	if err != nil {
+		if resp.StatusCode == http.StatusNotFound {
+			return nil
+		}
+		return fmt.Errorf("Error issuing AzureRM delete request for Application Insights WebTest '%s': %+v", name, err)
+	}
+
+	return err
+}
+
+func extractGeoLocations(d *schema.ResourceData) []insights.WebTestGeolocation {
```

Comment:
for newer resources we're passing in the `[]interface` rather than deserializing within this method; since it could be reused - as such could we make this:

```suggestion
func expandApplicationInsightsWebTestGeoLocations(input []interface{}) []insights.WebTestGeolocation {
```

(this also renamed `extract` to `expand` to match the other resources and prefixes the method name since methods are shared throughout the package)